app/controllers/api/v1/contacts_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class ContactsController < Api::V1::BaseController
+      before_action only: %i[index] do
+        require_access_token %w[read]
+      end
+
+      before_action only: %i[create destroy] do
+        require_access_token %w[read write]
+      end
+
+      rescue_from ActiveRecord::RecordNotFound do
+        render json: I18n.t("api.endpoint_errors.aspects.not_found"), status: :not_found
+      end
+
+      def index
+        order = ["contact_id IS NOT NULL DESC", "profiles.first_name ASC", "profiles.last_name ASC",
+                 "profiles.diaspora_handle ASC"]
```

Comment:
I can see that.  Because of the nature of the returns this may be something in the service that is only used by us versus something I'm going to graft into the place where I originally pulled the code idea from if I were to do that.

Reply:
I think I'm going to move this to the AspectsMembershipService and I'm going to create a second method for retrieving a contact's aspects which I'll need for the User API.

Reply:
To me the dirty stuff to database dirty stuff goes in service, so we only have clean code in the controller. That's why I said that. But that was not blocking for me.